### PR TITLE
Nethack cross

### PR DESCRIPTION
**Pull request by lheckemann**
###### Motivation for this change
Allow cross-compiling nethack for a wider range of dungeon-crawling fun.

cc maintainer @abbradar 

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

Tested by building `pkgsCross.aarch64-multiplatform.nethack` on an x86_64 machine, copying the result to the community box, and running it.

### CODE REVIEW

**Comment by lheckemann** [comment]
@grahamcofborg build pkgsCross.raspberryPi.nethack pkgsCross.aarch64-multiplatform.nethack nethack nethack-qt

File: pkgs/games/nethack/default.nix [hunk 1]
```diff
@@ -64,12 +65,20 @@ in stdenv.mkDerivation rec {
...
+    ${lib.optionalString qtMode ''
     sed \
       -e 's,^QTDIR *=.*,QTDIR=${qt5.qtbase.dev},' \
       -e 's,CFLAGS.*QtGui.*,CFLAGS += `pkg-config Qt5Gui --cflags`,' \
       -e 's,CFLAGS+=-DCOMPRESS.*,CFLAGS+=-DCOMPRESS=\\"${gzip}/bin/gzip\\" \\\
         -DCOMPRESS_EXTENSION=\\".gz\\",' \
       -i sys/unix/hints/linux-qt4
+    ''}
+    ${lib.optionalString (stdenv.buildPlatform != stdenv.hostPlatform) ''
+    ${buildPackages.perl}/bin/perl -p \
```
Comment (Mic92) [inline pkgs/games/nethack/default.nix]: Can you add a comment what you do here?

**Comment by lheckemann** [comment]
@grahamcofborg build nethack-qt

**Comment by lheckemann** [comment]
@GrahamcOfBorg build nethack-qt nethack-x11 nethack

**Comment by lheckemann** [comment]
@grahamcofborg build pkgsCross.raspberryPi.nethack pkgsCross.aarch64-multiplatform.nethack

**Comment by lheckemann** [comment]
I think this is a success now? >.< so many interleaved borg notifications

**Comment by grahamc** [comment]
@lheckemann maybe re-trigger the build now, I hid all the comments.

**Comment by lheckemann** [comment]
@GrahamcOfBorg build pkgsCross.raspberryPi.nethack pkgsCross.raspberryPi.nethack-x11 nethack-x11 nethack-qt nethack

**Comment by lheckemann** [comment]
OK, nethack-x11 fails to cross-compile because of failed dependencies, so let's try that again without x11.

**Comment by lheckemann** [comment]
@GrahamcOfBorg build pkgsCross.raspberryPi.nethack nethack-x11 nethack-qt nethack

**Comment by lheckemann** [comment]
The darwin failure is a problem in cross-building stdenv as well as an issue with nethack-qt that doesn't seem to be introduced by this, so we should be good to merge!

**Pull request merged**